x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/actions/info_button.tsx

```diff
@@ -0,0 +1,135 @@
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiPortal,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { Component, Fragment } from 'react';
+import { UISession } from '../../types';
+import { TableText } from '../';
+import { CodeEditor } from '../../../../../../../../src/plugins/kibana_react/public';
+import './info_button.scss';
+
+interface Props {
+  searchSession: UISession;
+}
+
+interface State {
+  isLoading: boolean;
+  isFlyoutVisible: boolean;
+  calloutTitle: string;
+}
+
+export class InfoButton extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isLoading: false,
+      isFlyoutVisible: false,
+      calloutTitle: 'Search Session Info',
+    };
+
+    this.closeFlyout = this.closeFlyout.bind(this);
+    this.showFlyout = this.showFlyout.bind(this);
+  }
+
+  public renderInfo() {
+    return (
+      <Fragment>
+        <CodeEditor
+          languageId="json"
+          value={JSON.stringify(this.props.searchSession.initialState, null, 2)}
+          onChange={() => {}}
+          options={{
+            readOnly: true,
+            lineNumbers: 'off',
+            fontSize: 12,
+            minimap: {
+              enabled: false,
+            },
+            scrollBeyondLastLine: false,
+            wordWrap: 'on',
+            wrappingIndent: 'indent',
+            automaticLayout: true,
+          }}
+        />
+      </Fragment>
+    );
+  }
+
+  public render() {
+    let flyout;
+
+    if (this.state.isFlyoutVisible) {
+      flyout = (
+        <EuiPortal>
+          <EuiFlyout
+            ownFocus
+            onClose={this.closeFlyout}
+            size="s"
+            aria-labelledby="flyoutTitle"
+            data-test-subj="searchSessionsFlyout"
+            className="searchSessionsFlyout"
+          >
+            <EuiFlyoutHeader hasBorder>
+              <EuiTitle size="m">
+                <h2 id="flyoutTitle">{this.state.calloutTitle}</h2>
+              </EuiTitle>
+            </EuiFlyoutHeader>
+            <EuiFlyoutBody>
+              <EuiText>
+                <EuiText size="xs">
+                  <p>
+                    <FormattedMessage
+                      id="xpack.data.sessions.management.flyoutText"
+                      defaultMessage="Original configuration used to create the search session:"
+                    />
+                  </p>
+                </EuiText>
+                <EuiSpacer />
+                {this.renderInfo()}
+              </EuiText>
+            </EuiFlyoutBody>
+          </EuiFlyout>
+        </EuiPortal>
+      );
+    }
+
+    return (
+      <Fragment>
+        <TableText onClick={this.showFlyout}>
+          <FormattedMessage
+            id="xpack.data.mgmt.searchSessions.actionInfo"
+            aria-label="Show search session info"
+            defaultMessage="Info"
```

Comment:
Should this be 'Original configuration' to be clearer in what is to be shown?  Though I admit I'm not sure configuration is clearer/better than info.  'Request' perhaps?

Reply:
Internally this is called "initialState". I'm open to any name @gchaps suggests!

Reply:
I spoke with @mdefazio, and I agree that "info" is a little vague. It's also inconsistent with the other items in the menu, as it doesn't start with a verb.  

Michael pointed out that in other areas of Kibana, we use the word "Inspect" to look under the hood.  Also, for the menu, I think we need to specify what is being extended.  Here is a suggestion:

### Menu

Inspect
Extend expiration
Delete

### Flyout

Title: Inspect search session

Options for intro text:

The request used to create this search session
Configuration for this search session

Reply:
Perfect! Thank you!